feature/KAS-3004 Keep capitalization of Raad van State

mdv-professional opened the pull request:
[Jira](https://kanselarij.atlassian.net/browse/KAS-3004)
[PR/KAS-3004/project](https://github.com/kanselarij-vlaanderen/kaleidos-project/pull/246)

**Issues**
1. In de *auk-css-additions* zit een helper `auk-u-text-capitalize`. Alleen deze blijkt niet te werken voor pills. `inline-flex` `display`-css-setting lijkt (als enige optie) de `::first-letter`-selector niet te gebruiken. Ik kan dit dus oplossen met de pill-content te wrappen in een span (gewoon inline). Ik zou dit meteen generieker kunnen oplossen door deze in Auk::Pill te zetten en een `@capitalize`-attribuut, al lijkt dit me ook niet helemaal ideaal.
2. Andere types hebben ook inconsistent hoofdletter gebruik. vb. subcase-types "*I*n voorbereiding" vs. "*b*ekrachtiging Vlaams Parlement".
3. De behoofdletteringsverschillen worden zichtbaar waar er geen `auk-u-text-capitalize` wordt gebruikt (nl. in `PowerSelect`). Ik zou hier ook een `@capitalize`-argument aan kunnen toevoegen.

Commit:
KAS-3004 keep capitalization of Raad van State

MikiDi commented:
`subcaseName.toLowerCase` zal een hack geweest zijn voor een eerdere vraag "doe die hoofdletter aan het begin van de zin weg aub".
Nu vraagt men "waar zijn die hoofdletters in Raad van State"?
Graag de bron voor deze strings (in de database?) aanpassen zodat ze geen hoofdletter hebben aan het begin van de zin, maar wel andere hoofdletters behouden. De string-operaties in de frontend kunnen dan verwijderd worden.

Commit:
KAS-3004 capitalize where appropriate
> keep capitalization of Raad van State

Commit:
KAS-3004 capitalize using css if possible
> keep capitalization of Raad van State

Commit:
Merge branch 'development' into feature/KAS-3004-capitalization-of-Raad-van-State

ghost commented:
Het verwijderen van de hoofdletters gaf weer nieuwe issues. Ik had hier graag advies over. Vandaar dat ik het in draft heb gezet.

brenner-company commented:
@MikiDi Wat is jouw mening over een eventuele `@capitalize` argument bij componenten?

Ik zou eerder neigen naar het correct vormen van de tekst (hoofdletter of niet) voor het meegeven aan een component. In het verleden heb ik ook al af en toe gelijkaardige discussies gehad met collega's (bij andere klanten) ivm het aanpassen van hoofdletters/eerste letters en toen is quasi altijd de conclusie er uit gekomen dat capitalization via CSS vaak flaky/situationeel is (zie voorbeeld `auk-pill`) en eerder moet gebruikt worden als er geen ander deftig alternatief voor handen is. Nu, ik wil dit zeker nog eens opnieuw bespreken/bekijken indien nodig.

MikiDi commented:
> capitalization via CSS vaak flaky/situationeel is (zie voorbeeld auk-pill)

Ik volg helemaal de redenering van @brenner-company . Voor specifieke componenten kan dat iets zijn, maar voor génerieke componenten geen CSS-capitalization.

@mdv-professional , als we er van uit gaan dat we voor het volgende gaan:
> het correct vormen van de tekst (hoofdletter of niet) voor het meegeven aan een component.

Aan welke data zijn er dan nog wijzigingen nodig om dit af te kunnen maken? Je mag deze wijzigingen dan meteen toevoegen aan de hierbij behorende backend PR.

ghost commented:
@MikiDi Ik heb gekeken naar enkele van de skos:Concepts in de databank maar dit lijkt te ver te leiden. Bijna allemaal starten ze met een hoofdletter.
Het enige andere skos:Concept dat ik in de frontend als fout behoofdletterd zie zijn de (dossier:)Procedurestap-types.

#### Conclusie
Na jullie commentaar, doe ik het volgende:
- **data**: ik corrigeer de volgende types binnen dit issue:
  - Procedurestap-namen (al gebeurd)
  - Procedurestap-types

- **weergave** binnen generieke componenten:
  - css-class vervangen door capitalize helper
  - Specifiek voor de `PowerSelect`: Het `@propertyName`-argument vervangen met een `yield`-blok met een `capitalize`-helper

Andere types lijken me voorlopig te veel werk om allemaal te wijzigen.

Laat maar weten als ge niet akkoord zijt.

Commit:
KAS-3004 capitalize using helper
reason: flex-inline ignores css class
> keep capitalization of Raad van State

Commit:
KAS-3004 use yield block to capitalize in ModelSelector
> keep capitalization of Raad van State

Commit:
KAS-3004 use capitalize helper to capitalize in ModelSelector
reason: otherwise ModelManager should accept yield block for capitalization
> keep capitalization of Raad van State

Commit:
Revert "KAS-3004 use capitalize helper to capitalize in ModelSelector"

This reverts commit 86dc773d7f267c47204799d13d50310965e28d00.

Commit:
KAS-3004 pass yield block to capitalize to ModelSelector in ModelManager
reason: otherwise ModelManager should accept yield block for capitalization
> keep capitalization of Raad van State

Commit:
KAS-3004 remove extraneous await
    > keep capitalization of Raad van State

MikiDi committed:
Merge branch 'development' into feature/KAS-3004-capitalization-of-Raad-van-State

ValenberghsSven committed:
KAS-3004 remove double css and use helper instead of css
